app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -13,21 +13,46 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
     function ActiveFiltersLink($scope, ngModel) {
         $scope.uiFilters = {};
         $scope.savedSearch = null;
+        $scope.collection = PostFilters.getModeEntity('collection');
         var originalSavedSearch;
         $scope.removeFilter = removeFilter;
         $scope.transformFilterValue = transformFilterValue;
         $scope.removeSavedSearch = removeSavedSearch;
+        $scope.removeCollection = removeCollection;
         $scope.showSaveSavedSearchButton = showSaveSavedSearchButton;
         $scope.userCanUpdateSavedSearch = false;
 
         activate();
-        $scope.$on('savedSearch:update', function () {
-            handleFiltersUpdate(PostFilters.getActiveFilters(PostFilters.getFilters()), null, true);
-        });
 
         $scope.isArray = angular.isArray;
 
         function activate() {
+            /**
+             * we need to listen for saved searches and collections update events
+             * so that we are notified when a user changes a collection or
+             * saved search
+             */
+            $scope.$on('savedSearch:update', function () {
+                handleFiltersUpdate(PostFilters.getActiveFilters(PostFilters.getFilters()), null, true);
+            });
+            $scope.$on('collection:update', function () {
+                handleFiltersUpdate(PostFilters.getActiveFilters(PostFilters.getFilters()), null, true);
+            });
+            /**
+             * we don't have filters set when we are on a collection
+             * so we need to watch for collection mode changes specifically instead of relying on the filters
+             * changing like we do for saved searches
+             */
+            $scope.$watch(function () {
+                return PostFilters.getModeEntity('collection');
```

Comment:
Does this still work if we just watch getModeId? I assume comparing `int`s is lower overhead than comparing objects

Reply:
yea, that would work if I watch getModeId and then just go and run getModeEntity('collection') . I'll switch it